hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/GCMergedRegionsProcedure.java

```diff
@@ -88,7 +87,7 @@ protected Flow executeFromState(MasterProcedureEnv env, GCMergedRegionsState sta
         setNextState(GCMergedRegionsState.GC_REGION_EDIT_METADATA);
         break;
       case GC_REGION_EDIT_METADATA:
-        MetaTableAccessor.deleteMergeQualifiers(env.getMasterServices().getConnection(), mergedChild);
+        env.getAssignmentManager().getRegionStateStore().deleteMergeQualifiers(mergedChild);
```

Comment:
Is it possible NPE in this long call chain?

Reply:
In real deploy, no. In tests we could have null assignmentmanager or regionstatestore, but as the above pre commit is fine so I do not think it is a problem here.